packages/ts/generator-typescript-cli/src/index.ts

```diff
@@ -46,7 +46,7 @@ const logger = new LoggerFactory({ verbose });
 const io = new GeneratorIO(outputDir, logger);
 
 const resolvedPlugins = await Promise.all(Array.from(new Set(plugins), (pluginPath) => io.loadPlugin(pluginPath)));
-const generator = new Generator(resolvedPlugins, logger);
+const generator = new Generator(resolvedPlugins, logger, outputDir);
```

Comment:
Let's create a `Context` data structure that contains `outputDir` and `logger` to make it easier to change the public API if it is necessary